life_expectancy/main.py

```diff
@@ -5,16 +5,34 @@
 import pandas as pd
 
 from life_expectancy.loading_saving import save_data, load_data
-from .data_cleaning import clean_data
+from life_expectancy.data_cleaning import clean_data_csv, clean_data_json
+from life_expectancy.country import Country
+
 
 CURRENT_FILEPATH = Path(__file__).parent.resolve()
 
+clean_function_to_apply = {
+    ".csv": clean_data_csv,
+    ".tsv": clean_data_csv,
+    ".json": clean_data_json
+    }
+
+def main(input_data_path: Path, output_data_path: Path, country: Country = Country["PT"])\
+    -> pd.DataFrame:
+    """main function that reads an inout file, cleans and saves it:
+
+    param: file_path: Path to the input file
+    (currently accepts files of type tsv, csv and json. The file can be compressed.)
+
+    output: file_path:  filepath of the file cleaned dataframe saved as a csv
+    """
+    life_expectancy_raw, filetype = load_data(input_data_path)
 
-def main(input_data_path: Path, output_data_path: Path, country: str) -> pd.DataFrame:
-    """main function: ....."""
-    eu_life_expectancy_raw = load_data(input_data_path)
-    eu_life_expectancy_filtered = clean_data(eu_life_expectancy_raw, country=country)
-    return save_data(eu_life_expectancy_filtered, output_data_path)
+
+    life_expectancy_filtered = clean_function_to_apply[filetype]\
+        (life_expectancy_raw, country=country.value)
```

Comment:
Giving the argument country like this would it not be possible to pass non-sense values (i.e. those that you remove in the class method in the enum)? In clean_data you do not check if the value is in the clean list.

Reply:
Estás a sugerir tipo fazer um raise caso o country de input não esteja na lista de paises aceite?

Reply:
Sim, no inicio do código posso chamar este class method para limpar a lista do country. 
On it.